[DO NOT MERGE] Overview of removing problematic whitelist/blacklist terminology

sarayourfriend opened the pull request:
#### Changes proposed in this Pull Request

* Change instances of the words "whitelist" and "blacklist" to alternative (often times more descriptive) terms https://github.com/Automattic/wp-calypso/issues/43302

#### Testing instructions

* All unit tests should pass
* todo: potentially manual testing is necessary

Fixes https://github.com/Automattic/wp-calypso/issues/43302

sarayourfriend committed:
Update: Rename "easy to solve" instances of "whitelist"

sarayourfriend committed:
Update: Rename "easy to solve" instances of "blacklist"

saramarcondes commented on client/lib/plugins/utils.js:
```diff
@@ -50,6 +50,7 @@ function filterNoticesBy( site, pluginSlug, log ) {
 	return false;
 }
 
+// @todo(saramarcondes) Is this safe to rename? Would plugins potentially depend on this function's current name?
```
@kwight I noticed you made the most recent "substantial" (non-formatting/chore) change to this file. Do you happen to know the answer to my question in this comment? Is it safe to rename this `whiteListPluginData` function?

kwight replied:
From what I can tell, the only two uses are here in this file: the definition, and a single call. The tests will have to be updated, but other than that, I think it's good to go!

saramarcondes replied:
Right on, thanks so much!

sarayourfriend committed:
Update: Rename Jetpack Protect Whitelist to "allowed IPs" where possible

Also removes the `protect__add-to-whitelist` class from the "add to
list" button. I couldn't find this class defined in any stylesheets
so I think it was just unused.

sarayourfriend committed:
Update: Rename whiteListPluginData to getAllowedPluginData

sarayourfriend committed:
Rename folder

simison commented on client/landing/gutenboarding/onboarding-block/acquire-intent/site-title.tsx:
```diff
@@ -50,7 +50,7 @@ const SiteTitle: React.FunctionComponent< Props > = ( { onSubmit } ) => {
 					autoComplete="off"
 					autoCorrect="off"
-					data-hj-whitelist
+					data-hj-whitelist // @todo(saramarcondes) where is this used?
```
This is to allow Hotjar.com show what user types into this field and unfortunately we cannot modify it.

https://help.hotjar.com/hc/en-us/articles/115015563287-How-to-Whitelist-Elements-and-Keystrokes

saramarcondes replied:
Ah okay, thanks for letting me know. I’ll add a note in a comment!

saramarcondes replied:
I reached out to HotJar yesterday and got a response from them that they're working on this internally. They're going to let me know once the data attribute has been updated to be able to be changed to something different.

simison replied:
Excellent! Thanks for checking with them.

p-jackson replied:
Looks like the new attribute name is `data-hj-allow`.
https://help.hotjar.com/hc/en-us/articles/115015563287-How-to-Whitelist-Elements-and-Keystrokes

Updating eslint exception here: #46620

sarayourfriend committed:
Update: Add note for HotJar attributes

sarayourfriend committed:
Update: Rename isAuthorNameBlocked to avoid confusing "block" terms

sarayourfriend committed:
Update: Rename isEmailBlocklisted to avoid confusing "blocklist" term

sarayourfriend committed:
Update: Rename isSiteBlocklistedError to avoid confusing "blocklist" term

sarayourfriend committed:
Update: Rename several areas to avoid confusing "blocklist" term

saramarcondes commented on client/my-sites/site-settings/form-discussion.jsx:
```diff
@@ -525,22 +525,22 @@ class SiteSettingsFormDiscussion extends Component {
 					autoCapitalize="none"
-					onClick={ eventTracker( 'Clicked Blacklist Field' ) }
-					onKeyPress={ uniqueEventTracker( 'Typed in Blacklist Field' ) }
+					onClick={ eventTracker( 'Clicked Disallowed Comments Field' ) }
+					onKeyPress={ uniqueEventTracker( 'Typed in Disallowed Comments Field' ) }
```
Discussion about whether this is safe: p5uIfZ-9OH-p2 (it sounds like it is!)

a8ci18n commented on client/lib/domains/registration/availability-messages.js:
```diff
@@ -254,7 +254,7 @@ function getAvailabilityNotice( domain, error, errorData ) {
 			} else {
 				message = translate(
-					'Domain cannot be mapped to a WordPress.com blog because of blacklisted term.'
+					'Domain cannot be mapped to a WordPress.com blog because of disallowed term.'
```
:ok: This change will be queued for retranslation. We'll use the existing translations in the meantime.

sarayourfriend committed:
Correct test assertions

saramarcondes commented:
The JS lint job is failing due to there being too high a number of JS files being passed to `eslint` in the `FILES_TO_LINT` variable. Not sure what to do about that but I've opened a ticket for Team Calypso to address the issue: https://github.com/Automattic/wp-calypso/issues/43631

gwwar commented on client/blocks/reader-feed-header/index.jsx:
```diff
@@ -13,7 +13,7 @@ import config from 'config';
  */
 import { Card } from '@automattic/components';
 import ReaderFollowButton from 'reader/follow-button';
-import { isAuthorNameBlocked } from 'reader/lib/author-name-blocklist';
```
Is blocklist troublesome here?

saramarcondes replied:
I suppose not, it's probably one of the cases where `blocklist` is pretty clear actually. I'll revert this one.

gwwar commented on client/jetpack-connect/connection-notice-types.js:
```diff
@@ -26,7 +26,7 @@ export const INVALID_PERMISSIONS = 'invalidPermissions';
 export const RETRYING_AUTH = 'retryingAuth';
 export const SECRET_EXPIRED = 'secretExpired';
-export const SITE_BLACKLISTED = 'siteBlacklisted';
+export const SITE_BLOCKED = 'siteBlacklisted';
```
```suggestion
export const SITE_BLOCKED = 'siteBlocked';
```

gwwar reviewed (commented):
If we'd like manual testing, I'd recommend splitting this out to a few PRs. Maybe one for Jetpack notices, and the rest a reasonable number of files per a certain set of folders/functionality.

sarayourfriend committed:
Update client/jetpack-connect/connection-notice-types.js

Co-authored-by: Kerry Liu <gwwar@users.noreply.github.com>

saramarcondes commented:
@gwwar Thanks for the review and suggestions! I'd definitely prefer manual testing. Agreed that splitting it makes sense for that. I'll get split PRs open either today or tomorrow.
